components/azure-iot/sample/Makefile

```diff
@@ -0,0 +1,8 @@
+#
+# This is a project Makefile. It is assumed the directory this Makefile resides in is a
+# project subdirectory.
+#
+
+PROJECT_NAME := esp32
```

Comment:
the project name should be set to something related to azure-iot (such as `azure-iot-sample`).